aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -99,278 +108,275 @@ The preceding command adds the [aspnet-codegenerator scaffolding tool](xref:fund
+With EF Core, data access is performed using a model. A model is made up of entity classes and a context object that represents a session with the database. The context object allows querying and saving data. The database context is derived from [Microsoft.EntityFrameworkCore.DbContext](xref:Microsoft.EntityFrameworkCore.DbContext) and specifies the entities to include in the data model.
 
-  You can ignore that warning, it will be fixed in a later tutorial.
+Scaffolding creates the *Data/MvcMovieContext.cs* database context class:
 
-[!INCLUDE [more information on the PMC tools for EF Core](~/includes/ef-pmc.md)]
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/zDocOnly/MvcMovieContext.cs?name=snippet)]
 
-# [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
+The preceding code creates a [DbSet\<Movie>](/dotnet/api/microsoft.entityframeworkcore.dbset-1) property that represents the movies in the database.
 
-[!INCLUDE [more information on the CLI for EF Core](~/includes/ef-cli.md)]
+ASP.NET Core is built with [dependency injection (DI)](xref:fundamentals/dependency-injection). Services, such as the database context, must be registered with DI in `Startup`. Components that require these services are provided these services via constructor parameters.
 
-Run the following .NET Core CLI commands:
+In the *Controllers/MoviesController.cs* file, the constructor uses [Dependency Injection](xref:fundamentals/dependency-injection) to inject the `MvcMovieContext` database context into the controller. The database context is used in each of the [CRUD](https://wikipedia.org/wiki/Create,_read,_update_and_delete) methods in the controller.
 
-```dotnetcli
-dotnet ef migrations add InitialCreate
-dotnet ef database update
-```
+Scaffolding generated the following highlighted code in `Startup.ConfigureServices`:
 
-* `ef migrations add InitialCreate`: Generates an *Migrations/{timestamp}_InitialCreate.cs* migration file. The `InitialCreate` argument is the migration name. Any name can be used, but by convention, a name is selected that describes the migration. Because this is the first migration, the generated class contains code to create the database schema. The database schema is based on the model specified in the `MvcMovieContext` class (in the *Data/MvcMovieContext.cs* file).
+# [Visual Studio](#tab/visual-studio)
 
-* `ef database update`: Updates the database to the latest migration, which the previous command created. This command runs the `Up` method in the *Migrations/{time-stamp}_InitialCreate.cs* file, which creates the database.
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Startup.cs?name=snippet_ConfigureServices&highlight=5-6)]
```

Comment:
Format code to prevent horizontal scroll bar on tablets 85 characters wide. You can simulate tablets by viewing in a browser of the maximum width that doesn't have left or right panes (TOC's).. 

![Uploading image.png…]()

Reply:
![image](https://user-images.githubusercontent.com/3605364/118339627-1b000b00-b4b5-11eb-8591-b251a24dd218.png)

Reply:
two spaces should work

Reply:
Get into the habit of testing code formatting to prevent horiz scroll bar.